tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -503,17 +503,55 @@ where
         Self::new_with_connection_cache(rpc_client, websocket_url, config, connection_cache).await
     }
 
+    /// Try to create LeaderTpuService
+    /// Retries until successful or timeout is reached
+    async fn try_create_leader_tpu_service(
+        rpc_client: Arc<RpcClient>,
+        websocket_url: &str,
+        exit: Arc<AtomicBool>,
+        timeout_seconds: u64,
+    ) -> Result<LeaderTpuService> {
```

Comment:
You can actually change the type of `self` to be an `Arc<T>`.

```suggestion
    /// Try to create LeaderTpuService
    /// Retries until successful or timeout is reached
    async fn create_leader_tpu_service(
        self: Arc<Self>,
        websocket_url: &str,
        exit: Arc<AtomicBool>,
        timeout_seconds: u64,
    ) -> Result<LeaderTpuService> {
```

Allowing method call syntax for the code that operates on `Arc`s and wants to use them.

Also, I am not sure that adding a `try_` prefix to every function that returns a `Result` is more useful than not.
We are working in a language that has rich type information, and we generally assume that readers are able to see or infer the types.
`Result` is a very standard way of reporting errors, and error handling is ubiquitously
Meaning that if we actually properly follow this pattern, more than half of our functions will start with `try_`.

A few other types of "method receivers" are supported as well:

> If the type of the `self` parameter is specified, it is limited to types resolving to one generated by the following grammar (where `'lt` denotes some arbitrary lifetime):
> ```
> P = &'lt S | &'lt mut S | Box<S> | Rc<S> | Arc<S> | Pin<P>
> S = Self | P
> ```

See https://doc.rust-lang.org/reference/items/associated-items.html#methods

Reply:
These methods are part of the `TpuClient`, so changing `rpc_client: Arc<RpcClient>` to `self: Arc<Self>` doesn't really work

And sounds good on `try_` will update